code/app/src/main/java/com/example/goldencarrot/views/FacilityProfileActivity.java

```diff
@@ -0,0 +1,89 @@
+package com.example.goldencarrot.views;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.example.goldencarrot.R;
+import com.example.goldencarrot.data.model.user.UserFacility;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.squareup.picasso.Picasso;
+
+public class FacilityProfileActivity extends AppCompatActivity {
+    private static final String TAG = "FacilityProfileActivity";
+
+    private EditText nameEditText, locationEditText, descriptionEditText, contactInfoEditText;
+    private ImageView facilityImageView;
+    private Button saveButton;
+
+    private FirebaseFirestore db;
+    private FirebaseAuth auth;
+    private String facilityID;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_facility_profile);
+
+        // Initialize Firebase instances
+        db = FirebaseFirestore.getInstance();
+        auth = FirebaseAuth.getInstance();
+
+        // Initialize views
+        nameEditText = findViewById(R.id.nameEditText);
+        locationEditText = findViewById(R.id.locationEditText);
+        descriptionEditText = findViewById(R.id.descriptionEditText);
+        contactInfoEditText = findViewById(R.id.contactInfoEditText);
+        facilityImageView = findViewById(R.id.facilityImageView);
+        saveButton = findViewById(R.id.saveButton);
+
+        // Check if facilityID was passed in, if so load the profile for editing
+        facilityID = getIntent().getStringExtra("facilityID");
+        if (facilityID != null) {
+            loadFacilityProfile();
+        }
+
+        saveButton.setOnClickListener(view -> saveFacilityProfile());
+    }
+
+    private void loadFacilityProfile() {
+        DocumentReference docRef = db.collection("facilities").document(facilityID);
+        docRef.get().addOnSuccessListener(documentSnapshot -> {
+            UserFacility profile = documentSnapshot.toObject(UserFacility.class);
+            if (profile != null) {
+                nameEditText.setText(profile.getName());
```

Comment:
Why are we adding this document to firebase? this seems very similar to the user documents we already have

Reply:
we can maybe have organizer users have more fields? please do a POC first

having another event collection for facilities might overcomplicate things, we have been using user objects for all other views, methods